js/songs/edit.controller.js

```diff
@@ -0,0 +1,21 @@
+"use strict";
+
+(function(){
+  angular
+  .module("songs")
+  .controller("SongEditController", [
+    "SongFactory",
+    "$stateParams",
```

Comment:
in order to trigger a `redirect` with UI Router, we have load in a module `$state` as a dependency to our controller.